File: mobile/lib/ui/theme.dart
import 'package:flutter/material.dart';

/// Bảng màu lấy NGUYÊN VĂN từ web, không tự chọn lại.
///
/// App di động và web đặt món là hai cửa vào cùng một quán. Khách quét QR ở bàn rồi mở app phải
/// thấy cùng một nơi, không phải hai sản phẩm khác nhau — nên màu ở đây chép đúng biến CSS của
/// `frontend/src/components/customer/customer-menu.css` (chủ đề "Vị An"):
///
///     --brand-chestnut       #6f3d2c   nâu hạt dẻ, màu chính
///     --brand-chestnut-dark  #4c291f
///     --brand-brass          #a47834   đồng thau, màu nhấn
///     --brand-ivory          #fffaf5   nền
///     --brand-ink            #2f1d16   chữ
///     --brand-muted          #80685e   chữ phụ
///     --brand-clay-line      #ead8cd   viền
///     --brand-success        #2f7251
///     --brand-danger         #b13c32
///     --vian-card-bg         #f1dfd3
///
/// Bản trước dùng `colorSchemeSeed: Colors.deepOrange`, cho ra một dải cam Material tự sinh —
/// gần đúng tông ấm nhưng không phải màu của quán, và đó là chỗ "không đồng bộ" nhìn thấy ngay.
class MauQuan {
  const MauQuan._();

  static const chestnut = Color(0xFF6F3D2C);
  static const chestnutDark = Color(0xFF4C291F);
  static const brass = Color(0xFFA47834);
  static const ivory = Color(0xFFFFFAF5);
  static const ink = Color(0xFF2F1D16);
  static const muted = Color(0xFF80685E);
  static const clayLine = Color(0xFFEAD8CD);
  static const success = Color(0xFF2F7251);
  static const danger = Color(0xFFB13C32);
  static const cardBg = Color(0xFFF1DFD3);
  static const beige = Color(0xFFF8EEE5);
}

/// Bo góc lấy từ `--radius-*` của web: thẻ món web dùng 20px.
class BoGoc {
  const BoGoc._();

  static const nho = 8.0;
  static const vua = 12.0;
  static const lon = 16.0;
  static const the = 20.0;
}

ThemeData chuDeQuan() {
  const scheme = ColorScheme(
    brightness: Brightness.light,
    primary: MauQuan.chestnut,
    onPrimary: Colors.white,
    primaryContainer: MauQuan.beige,
    onPrimaryContainer: MauQuan.chestnutDark,
    secondary: MauQuan.brass,
    onSecondary: Colors.white,
    secondaryContainer: MauQuan.cardBg,
    onSecondaryContainer: MauQuan.chestnutDark,
    error: MauQuan.danger,
    onError: Colors.white,
    surface: MauQuan.ivory,
    onSurface: MauQuan.ink,
    surfaceContainerHighest: MauQuan.beige,
    onSurfaceVariant: MauQuan.muted,
    outline: MauQuan.clayLine,
  );

  return ThemeData(
    useMaterial3: true,
    colorScheme: scheme,
    scaffoldBackgroundColor: MauQuan.ivory,
    // KHÔNG kèm tệp font. "Be Vietnam Pro" của web sẽ làm APK nặng thêm vài trăm KB cho mỗi độ
    // đậm, và font hệ thống của Android hiển thị tiếng Việt có dấu đầy đủ. Ghi ra đây để đây là
    // một lựa chọn, không phải một thiếu sót.
    appBarTheme: const AppBarTheme(
      backgroundColor: MauQuan.ivory,
      foregroundColor: MauQuan.ink,
      elevation: 0,
      scrolledUnderElevation: 0.5,
      centerTitle: false,
    ),
    cardTheme: CardThemeData(
      color: Colors.white,
      elevation: 0,
      margin: EdgeInsets.zero,
      shape: RoundedRectangleBorder(
        borderRadius: BorderRadius.circular(BoGoc.the),
        side: const BorderSide(color: MauQuan.clayLine),
      ),
    ),
    filledButtonTheme: FilledButtonThemeData(
      style: FilledButton.styleFrom(
        backgroundColor: MauQuan.chestnut,
        foregroundColor: Colors.white,
        padding: const EdgeInsets.symmetric(horizontal: 20, vertical: 14),
        shape: RoundedRectangleBorder(
            borderRadius: BorderRadius.circular(BoGoc.vua)),
      ),
    ),
    outlinedButtonTheme: OutlinedButtonThemeData(
      style: OutlinedButton.styleFrom(
        foregroundColor: MauQuan.chestnut,
        side: const BorderSide(color: MauQuan.clayLine),
        padding: const EdgeInsets.symmetric(horizontal: 20, vertical: 14),
        shape: RoundedRectangleBorder(
            borderRadius: BorderRadius.circular(BoGoc.vua)),
      ),
    ),
    textButtonTheme: TextButtonThemeData(
      style: TextButton.styleFrom(foregroundColor: MauQuan.chestnut),
    ),
    inputDecorationTheme: InputDecorationTheme(
      filled: true,
      fillColor: Colors.white,
      border: OutlineInputBorder(
        borderRadius: BorderRadius.circular(BoGoc.vua),
        borderSide: const BorderSide(color: MauQuan.clayLine),
      ),
      enabledBorder: OutlineInputBorder(
        borderRadius: BorderRadius.circular(BoGoc.vua),
        borderSide: const BorderSide(color: MauQuan.clayLine),
      ),
    ),
    navigationBarTheme: const NavigationBarThemeData(
      backgroundColor: Colors.white,
      indicatorColor: MauQuan.beige,
      elevation: 3,
      labelTextStyle: WidgetStatePropertyAll(
        TextStyle(
            fontSize: 11, color: MauQuan.ink, fontWeight: FontWeight.w500),
      ),
    ),
    dividerTheme: const DividerThemeData(color: MauQuan.clayLine, thickness: 1),
    chipTheme: ChipThemeData(
      backgroundColor: MauQuan.beige,
      side: const BorderSide(color: MauQuan.clayLine),
      labelStyle: const TextStyle(fontSize: 11, color: MauQuan.muted),
      shape: RoundedRectangleBorder(
          borderRadius: BorderRadius.circular(BoGoc.nho)),
    ),
  );
}

File: mobile-rn/src/ui/theme.ts
import { StyleSheet } from 'react-native';

/**
 * Bảng màu lấy NGUYÊN VĂN từ web, không tự chọn lại.
 *
 * App di động và web đặt món là hai cửa vào cùng một quán. Khách quét QR ở bàn rồi mở app phải
 * thấy cùng một nơi, không phải hai sản phẩm khác nhau — nên màu ở đây chép đúng biến CSS của
 * `frontend/src/components/customer/customer-menu.css` (chủ đề "Vị An").
 */
export const MauQuan = {
  chestnut: '#6f3d2c',
  chestnutDark: '#4c291f',
  brass: '#a47834',
  ivory: '#fffaf5',
  ink: '#2f1d16',
  muted: '#80685e',
  clayLine: '#ead8cd',
  success: '#2f7251',
  danger: '#b13c32',
  cardBg: '#f1dfd3',
  beige: '#f8eee5',
  trang: '#ffffff',
} as const;

/** Bo góc lấy từ `--radius-*` của web: thẻ món web dùng 20px. */
export const BoGoc = { nho: 8, vua: 12, lon: 16, the: 20 } as const;

/**
 * Kiểu dùng chung thay cho `ThemeData` của Flutter.
 *
 * React Native không có tầng chủ đề áp xuống mọi widget, nên thứ thay thế phải là một bộ kiểu
 * được các màn hình dùng lại. Ghi ra đây thay vì để mỗi màn hình tự đặt màu: nhân bản màu là cách
 * chắc chắn để hai màn hình lệch nhau vài sắc độ mà không ai nhận ra cho tới lúc chụp màn hình.
 *
 * KHÔNG kèm tệp font. "Be Vietnam Pro" của web sẽ làm bản dựng nặng thêm cho mỗi độ đậm, và font
 * hệ thống hiển thị tiếng Việt có dấu đầy đủ. Ghi ra để đây là một lựa chọn, không phải thiếu sót.
 */
export const kieuChung = StyleSheet.create({
  man: { flex: 1, backgroundColor: MauQuan.ivory },
  than: { padding: 20, gap: 16 },
  tieuDe: { fontSize: 22, fontWeight: '600', color: MauQuan.ink },
  chu: { fontSize: 15, color: MauQuan.ink, lineHeight: 22 },
  chuPhu: { fontSize: 13, color: MauQuan.muted, lineHeight: 19 },
  nhan: { fontSize: 13, fontWeight: '500', color: MauQuan.muted, marginBottom: 6 },
  oNhap: {
    backgroundColor: MauQuan.trang,
    borderWidth: 1,
    borderColor: MauQuan.clayLine,
    borderRadius: BoGoc.vua,
    paddingHorizontal: 14,
    paddingVertical: 12,
    fontSize: 16,
    color: MauQuan.ink,
  },
  nutChinh: {
    backgroundColor: MauQuan.chestnut,
    borderRadius: BoGoc.vua,
    paddingHorizontal: 20,
    paddingVertical: 14,
    alignItems: 'center',
  },
  chuNutChinh: { color: MauQuan.trang, fontSize: 16, fontWeight: '600' },
  nutVien: {
    borderWidth: 1,
    borderColor: MauQuan.clayLine,
    borderRadius: BoGoc.vua,
    paddingHorizontal: 20,
    paddingVertical: 14,
    alignItems: 'center',
  },
  chuNutVien: { color: MauQuan.chestnut, fontSize: 16, fontWeight: '600' },
  nutTat: { opacity: 0.5 },
  the: {
    backgroundColor: MauQuan.trang,
    borderWidth: 1,
    borderColor: MauQuan.clayLine,
    borderRadius: BoGoc.the,
    padding: 16,
  },
});
